pkg/arvo/app/ping.hoon

```diff
@@ -355,48 +135,16 @@
   ::
   =^  cards  state
     ?:  ?=([%kick ?] q.vase)
-      :: NB: ames calls this on %born (with fail=%.n) and after not hearing STUN
-      ::  responses for more than ~s5 (with fail=%.y)
-      ::
-      ::  if %ping was turned off (due to a successfull STUN) but we failed
-      ::  to get a STUN response in time switch to %nat and start a ~s25 timer
-      ::
-      ::  if the %kick has fail=%.n (e.g. for every %born), the plan will remain
-      ::  unchanged, but we will innitiate a new round of %poke pings with
-      ::  increasing nonce.
-      ::
-      ::  if we get repeated [%stun fail=&], but we are already in either %nat
-      ::  or %pub, do nothing, since there are already timers in place to %ping
-      ::  repeatedly.
-      ::
-      =/  stun-failed=?  &(?=([%off ~] plan.state) =(+.q.vase %.y))
-      ?:  &(?=([%off ~] plan.state) =(+.q.vase %.n))
-        ::  ignore restarts if we were already STUNning, if ip:port changed
-        ::  %once will trigger one formal %ping
-        ::
-        `state
-      =?  plan.state  stun-failed
-        [%nat ~]
-      ?:  &(!stun-failed =(+.q.vase %.y))
-        `state
-      (kick:ships our.bowl now.bowl)
-    ?:  =(q.vase %stop)  :: NB: ames calls this on [%stun fail=%.n]
-      =.  plan.state  [%off ~]
-      (kick:ships our.bowl now.bowl)
-    ?:  &(=(q.vase %once) =(%off -.plan.state))  :: NB: ames calls this on %once
-      =.  plan.state  [%one ~]
-      (kick:ships our.bowl now.bowl)
-    ?:  =(q.vase %nat)
-      =.  plan.state  [%nat ~]
-      (kick:ships our.bowl now.bowl)
-    ?:  =(q.vase %no-nat)
-      =.  plan.state  [%pub ~]
-      (kick:ships our.bowl now.bowl)
+      =?  mode.state  =(+.q.vase %.y)
+        %formal
+      (ping (galaxy-for our.bowl bowl))
+    ::
+    ?:  |(=(q.vase %once) =(q.vase %stop))  :: NB: ames calls this on %once
+      =.  mode.state  %informal
+      (ping (galaxy-for our.bowl bowl))
```

Comment:
I wonder if we have a race condition here with `%once`, in the case where we have already sent a poke to the galaxy but it hasn't been acknowledged yet. Presumably the galaxy will hear from the correct lane in most cases, but it seems like it's possible for the poke to be sent on lane *A*, the lane to change to *B*, `%once to be injected but no-op (because we've already sent and the galaxy is unchanged), and then the ack to come back.

Online lane changes are presumably pretty rare (ISP reconfiguration, DHCP lease expiration, that sort of thing). I'll need to think about how to handle this scenario, and how important it is.